File: AddressBook/AddressBook/Dtos/Translation.cs
using AddressBook.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class TranslationRes
    {
        public int TranslationId { get; set; }
        public string? Code { get; set; }
        public string? Name { get; set; }

        // Foreign keys
        public int? CountryId { get; set; }
        public Country? Country { get; set; }
    }

    public class TranslationReqEdit
    {
        [Required]
        public string? Code { get; set; }
        [Required]
        public string? Name { get; set; }

        // Foreign keys
        public int? CountryId { get; set; }
    }

    public class TranslationReqSearch : PagedReq
    {
        public int? CountryId { get; set; }
    }
}

File: AddressBook/react-client/src/dtos/Translation.ts
import Common from "../utility/Common";
import { CountryRes } from "./Country";
import { PagedReq } from "./PagedReq";

export interface TranslationRes {
  translationId?: string;
  code?: string;
  name?: string;

  countryId?: string;
  country?: CountryRes;
}

export class TranslationReqEdit {
  code?: string = "";
  name?: string = "";

  countryId?: string = "";
  constructor(countryId?: string) {
    this.countryId = countryId;
  }
}

export class TranslationReqSearch extends PagedReq {
  countryId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {countryId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.countryId = countryId;
  }
}